src/ManagedServices/ManagedServices/Commands/RemoveAzureRmManagedServcicesAssignment.cs

```diff
@@ -0,0 +1,93 @@
+{
+    [Cmdlet(
+        VerbsCommon.Remove,
+        Microsoft.Azure.Commands.ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "ManagedServicesAssignment",
+        DefaultParameterSetName = DefaultParameterSet,
+        SupportsShouldProcess = true), OutputType(typeof(PSRegistrationDefinition))]
+    public class RemoveAzureRmManagedServcicesAssignment : ManagedServicesCmdletBase
+    {
+        protected const string DefaultParameterSet = "Default";
+        protected const string ByResourceIdParameterSet = "ByResourceId";
+        protected const string ByInputObjectParameterSet = "ByInputObject";
+
+        [Parameter(Position = 0, ParameterSetName = DefaultParameterSet, Mandatory = false, HelpMessage = "The scope where the registration assignment should be created.")]
+        [ScopeCompleter]
+        public string Scope { get; set; }
+
+        [Parameter(ParameterSetName = DefaultParameterSet, Mandatory = true, HelpMessage = "The registration assignment Id.")]
+        [ValidateNotNullOrEmpty]
+        public string Id { get; set; }
+
+        [Parameter(ParameterSetName = ByResourceIdParameterSet, Mandatory = true, ValueFromPipelineByPropertyName = true, HelpMessage = "The fully qualified rsource id of the registration assignment.")]
+        [ValidateNotNullOrEmpty]
+        public string ResourceId { get; set; }
+
+        [Parameter(Mandatory = true, ValueFromPipeline = true, HelpMessage = "The registration assignment object.", ParameterSetName = ByInputObjectParameterSet)]
+        [ValidateNotNull]
+        public PSRegistrationAssignment InputObject { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = "Run cmdlet in the background")]
+        public SwitchParameter AsJob { get; set; }
+
+        public override void ExecuteCmdlet()
+        {
+            string scope = null;
+            string assignmentId = null;
+
+            if (this.IsParameterBound(x => x.Id))
+            {
+                assignmentId = this.Id;
+                scope = this.Scope ?? this.GetDefaultScope();
+            }
+            else if (this.IsParameterBound(x => x.InputObject))
+            {
+                assignmentId = this.InputObject.Id.GetResourceName();
+                if (!ManagedServicesUtility.TryParseAssignmentScopeFromResourceId(this.InputObject.Id, out scope))
+                {
+                    throw new ApplicationException($"Unable to parse the scope from [{this.InputObject.Id}]");
+                }
+            }
+            else if (this.IsParameterBound(x => x.ResourceId))
+            {
+                assignmentId = this.ResourceId.GetResourceName();
+                if (!ManagedServicesUtility.TryParseAssignmentScopeFromResourceId(this.ResourceId, out scope))
+                {
+                    throw new ApplicationException($"Unable to parse the scope from [{this.ResourceId}]");
+                }
+            }
+
+            ConfirmAction(MyInvocation.InvocationName,
```

Comment:
Identify the type fo resource being removed, and identifying information - think of this as a way for a user to validate a script, or to check the effects of a complex command before executing